### PR TITLE
Run workflows only for PRs and master

### LINKED ISSUE #1980: Avoid running the same workflows for push & PR events
Currently many PRs run duplicated workflows, one for the PR and one for the push trigger.

To save resources this could be filtered to run the PR one's only? Push workflows are only needed on the default branch?

image

### PR DESCRIPTION
**Pull request by pat-s**
fix #1980

### CODE REVIEW

File: .woodpecker/docker.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+when:
+  event: [push, pull_request]
+  branch: [master, release/*]
```
Comment (anbraten) [inline .woodpecker/docker.yml:3]: I would still prefer the following syntax:

(will try if it actually works in a test repo)

```yml
when:
  - event: pull_request
  - event: push
    branch: [master, release/*]
```

**Comment by pat-s** [comment]
@anbraten I've tested your proposed syntax -> it works. Good to merge?

File: .woodpecker/docker.yml [hunk 2]
```diff
@@ -1,3 +1,8 @@
+when:
+  - event: pull_request
+  - event: push
+    branch: [master, release/*]
```
Comment (6543) [inline .woodpecker/docker.yml:4]:
```suggestion
    branch: [${CI_REPO_DEFAULT_BRANCH}, release/*]
```

File: .woodpecker/docs.yml [hunk 3]
```diff
@@ -1,3 +1,8 @@
+when:
+  - event: pull_request
+  - event: push
+    branch: [master, release/*]
```
Comment (6543) [inline .woodpecker/docs.yml:4]:
```suggestion
    branch: [${CI_REPO_DEFAULT_BRANCH}, release/*]
```

File: .woodpecker/test.yml [hunk 4]
```diff
@@ -1,3 +1,8 @@
+when:
+  - event: pull_request
+  - event: push
+    branch: [master, release/*]
```
Comment (6543) [inline .woodpecker/test.yml:4]:
```suggestion
    branch: [${CI_REPO_DEFAULT_BRANCH}, release/*]
```

File: .woodpecker/web.yml [hunk 5]
```diff
@@ -1,3 +1,8 @@
+when:
+  - event: pull_request
+  - event: push
+    branch: [master, release/*]
```
Comment (6543) [inline .woodpecker/web.yml:4]:
```suggestion
    branch: [${CI_REPO_DEFAULT_BRANCH}, release/*]
```

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
well to early ... I would say :/

did had to https://github.com/woodpecker-ci/woodpecker/commit/87cef4177c7d80af9170f73e89bd1041503f06fa

**Comment by 6543** [comment]
I need to look at why the env-var did not work

**Pull request merged**